### PR TITLE
Flow searchbar

### PR DESCRIPTION
**Pull request by jasonLaster**

### CODE REVIEW

File: src/components/Editor/SearchBar.js [hunk 1]
```diff
@@ -65,7 +67,7 @@ const SearchBar = React.createClass({
     }
   },
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps: any) {
```
Comment (clarkbw) [inline src/components/Editor/SearchBar.js]: You can use `Props` here. https://github.com/facebook/flow/blob/v0.35.0/lib/react.js#L38

File: src/components/Editor/SearchBar.js [hunk 2]
```diff
@@ -133,19 +135,21 @@ const SearchBar = React.createClass({
...
     const sourceText = this.props.sourceText;
     const count = countMatches(query, sourceText.get("text"));
     // eslint-disable-next-line react/no-did-update-set-state
     this.setState({ query, count, index: 0 });
     this.search(query);
   },
 
-  onChange(e) {
-    this.doSearch(e.target.value);
+  onChange(e: any) {
```
Comment (clarkbw) [inline src/components/Editor/SearchBar.js]: `SyntheticEvent` here?

Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: unfortunately that doesn't work for the eventTarget reason

File: src/components/Editor/SearchBar.js (hunk 1 above)
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: I don't have Props defined

File: src/components/Editor/SearchBar.js (hunk 2 above)
Comment (clarkbw) [inline src/components/Editor/SearchBar.js]: I thought you just need guard against the target value being null.

`if (e.target && e.target.value)`

Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: yeah, that didn't work...

**Comment by jasonLaster** [comment]
@clarkbw sounds good. I can leave this open for a bit

**Comment by clarkbw** [comment]
> @clarkbw sounds good. I can leave this open for a bit

Ok, lets go ahead with this.  Everything I found hasn't worked and this is a good step forward.

**Pull request merged**